Reader.cpp

```diff
@@ -34,7 +34,7 @@ Reader::Reader(std::string fname, size_t buffer_size)
 
 std::shared_ptr<Block> Reader::next(const std::string& outer_block_name)
 {
-    while (true) {
+    for (;;) {
```

Comment:
Maybe:

    std::shared_ptr<Block> block;
    do {
        m_mae_parser->whitespace();
        block = m_mae_parser->outerBlock();
    } while (block != nullptr && block->getName() != outer_block_name);
    return block;

Man, it seems like some of this stuff would be clearer if the readers consumed the whitespace at a lower level.

Reply:
+1 for the low level space consumption.

Also, this alternative looks better than the other one.